src/app/services/stack/stack.service.ts

```diff
@@ -1,9 +1,155 @@
+     * @returns {Observable<Stack>}
+     */
+  public getStacks(): Observable<Stack[]>  {
+      return this.http
+          .get(this.config.getAPIHostname() + '/stacks', {
+              responseType: 'json',
+          })
+          .pipe(
+              map((x) => {
+                  return <Stack[]>x['data'];
+              }),
+              catchError((err: HttpErrorResponse) => {
+                  return ErrorObservable.create(<StackError>err.status);
+              })
+          );
+  }
+
+    /**
+     * Creates a new stack.
+     *
+     * Note that the _64params field is a base64 encoded file.
+     * @returns {Observable<StackError>}
+     */
+  public deployStack(name: string, _64params: string): Observable<StackError> {
+        return this.http
+            .post(
+                this.config.getAPIHostname() +
+                '/stacks/',
+                { stackName: name, stackFile: _64params},
+                { responseType: 'json' }
+            )
+            .pipe(
+                map((x) => {
+                    return x;
+                }),
+                catchError((err: HttpErrorResponse) => {
+                    return ErrorObservable.create(<StackError>err.status);
+                })
+            );
+  }
+
+    /**
+     * Returns a list of tasks associated with the stack.
+     *
+     * @param {string} stack
+     * @returns {Observable<JSON>}
+     */
+  public getStackTasks(stack: string): Observable<JSON> {
+      return this.http
+          .get(this.config.getAPIHostname() + '/stacks/' + stack + '/tasks', {
+              responseType: 'json',
+          })
+          .pipe(
+              map((x) => {
+                  return x;
+              }),
+              catchError((err: HttpErrorResponse) => {
+                  return ErrorObservable.create(<StackError>err.status);
+              })
+          );
+  }
+
+    /**
+     * Returns a list of services associated with the stack.
+     *
+     * @param {string} stack
+     * @returns {Observable<JSON>}
+     */
+    public getStackServices(stack: string): Observable<JSON> {
+        return this.http
+            .get(this.config.getAPIHostname() + '/stacks/' + stack + '/services', {
+                responseType: 'json',
+            })
+            .pipe(
+                map((x) => {
+                    return x;
+                }),
+                catchError((err: HttpErrorResponse) => {
+                    return ErrorObservable.create(<StackError>err.status);
+                })
+            );
+    }
+
+    /**
+     * Updates an existing stack.
+     *
+     * Note that the _64params field is a base64 encoded file.
+     * @returns {Observable<StackError>}
+     */
+    public updateStack(name: string, _64params: string): Observable<StackError> {
+        return this.http
+            .put(
+                this.config.getAPIHostname() +
+                '/stacks/' + name,
+                { stackName: name, stackFile: _64params},
```

Comment:
The `stackName` will be ignored by the back-end, the spec has changed in the meantime.